### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by cisacke**
Super light testing setup using Jest - pretty much works out of the box. The only thing I had to configure was the css files - because we use CSS modules, we need to use [identity-obj-proxy](https://www.npmjs.com/package/identity-obj-proxy) which mocks classNames nicely.

### CODE REVIEW

**Comment by drewdrewthis** [comment]
@cisacke This LGTM. @jmcolella had a comment about using Chai?

**Comment by drewdrewthis** [comment]
@cisacke If we're going to use this for all the tests, can you please update the plopfile / templater?

**Comment by cisacke** [comment]
@drewdrewthis yup good call -- added to the plopfile
@jmcolella added Chai 🍵

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (jmcolella) [inline .babelrc]: is this file necessary? these presets are already set in the webpack config

File: src/atoms/Button/__tests__/button.spec.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import Button from '../index';
```
Comment (jmcolella) [inline src/atoms/Button/__tests__/button.spec.js]: this relative import should go after the absolute imports; the linter will complain

File: plop_templates/test_template.txt [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import sinon from 'sinon';
+import { expect } from 'chai';
+import { shallow } from 'enzyme';
+
```
Comment (jmcolella) [inline plop_templates/test_template.txt]: maybe add the import statement for the component? `import {{properCase name }} from {{type}}/{{properCase name}}`...or maybe not all components will have this specific path? just a thought

File: src/atoms/Button/__tests__/button.spec.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    };
+  });
+
+  it('is a <button> tag', () => {
+    const wrapper = shallow(<Button />);
+    expect(wrapper.type()).to.equal('button');
+  });
+
+  it('adds the style prop as a className', () => {
+    const props = assign({}, defaultProps, {style: 'info'});
```
Comment (jmcolella) [inline src/atoms/Button/__tests__/button.spec.js]: slight wrench here...this test will fail with my PR. `style` is a reserved word in React and must have an object as its value. The linter complains about this and I agree, so I changed this prop value to `kind`. We need to once and for all have a team meeting about how to handle variant prop names

File: plop_templates/test_template.txt (hunk 3 above)
Comment (patrickkim) [inline plop_templates/test_template.txt]: Yes, I think this would help.

File: src/atoms/Button/__tests__/button.spec.js (hunk 4 above)
Comment (patrickkim) [inline src/atoms/Button/__tests__/button.spec.js]: make it happen @jmcolella

Comment (cisacke) [inline src/atoms/Button/__tests__/button.spec.js]: @jmcolella ahh ok - I'll hold off merging this until your PR is ready, and then I'll switch over to using `kind` or whatever we decide to use

File: plop_templates/test_template.txt (hunk 3 above)
Comment (cisacke) [inline plop_templates/test_template.txt]: 👍  good call - I think it's safe to assume they'll have the same relative path

File: .babelrc (hunk 1 above)
Comment (cisacke) [inline .babelrc]: @jmcolella yeah I'm not sure to be honest - when I remove the `.babelrc` the js doesn't compile when the tests run but I don't know why. Although I did discover that you can put the presets in `package.json` instead of creating a .babelrc so I've done that instead

File: src/atoms/Button/__tests__/button.spec.js (hunk 4 above)
Comment (patrickkim) [inline src/atoms/Button/__tests__/button.spec.js]: > variant

**Comment by cisacke** [comment]
@patrickkim resolved conflicts & changed button to use variant so tests will pass - lmk if this looks good to merge!
